src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -149,6 +155,127 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
+
+                    LockFile newAssetsFile = null;
+                    if (_request.ExistingLockFile != null)
+                    {
+                        newAssetsFile = _request.ExistingLockFile.Clone();
+                        newAssetsFile.LogMessages = logs;
+                    }
+
+                    if (cacheFile != null)
+                    {
+                        cacheFile.Success = _success;
+                    }
+
+                    return new RestoreResult(
+                        success: _success,
+                        restoreGraphs: new List<RestoreTargetGraph>(),
+                        compatibilityCheckResults: new List<CompatibilityCheckResult>(),
+                        msbuildFiles: new List<MSBuildOutputFile>(),
+                        lockFile: newAssetsFile,
+                        previousLockFile: _request.ExistingLockFile,
+                        lockFilePath: _request.ExistingLockFile?.Path,
+                        cacheFile: cacheFile,
+                        cacheFilePath: _request.Project.RestoreMetadata.CacheFilePath,
+                        packagesLockFilePath: nuGetLockFilePath,
+                        packagesLockFile: nuGetLockFile,
+                        projectStyle: _request.ProjectStyle,
+                        elapsedTime: restoreTime.Elapsed);
+                }
+
+                var isLockFileValid = false;
+
+                // read packages.lock.json file if exists and ReevaluateNuGetLockFile is not set to true
+                if (!_request.Project.RestoreMetadata.RestoreLockProperties.ReevaluateNuGetLockFile &&
+                    File.Exists(nuGetLockFilePath))
+                {
+                    nuGetLockFile = PackagesLockFileFormat.Read(nuGetLockFilePath, _logger);
+
+                    if (dgSpec != null && nuGetLockFile.Targets.Count > 0)
+                    {
+                        // check if lock file is out of sync with project data
+                        isLockFileValid = PackagesLockFileUtilities.IsLockFileStillValid(dgSpec, nuGetLockFile);
+
+                        if (isLockFileValid)
+                        {
+                            // pass lock file details down to generate restore graph
+                            foreach (var target in nuGetLockFile.Targets)
+                            {
+                                var libraries = target.Dependencies
+                                    .Where(dep => dep.Type != PackageInstallationType.Project)
+                                    .Select(dep => new LibraryIdentity(dep.Id, dep.ResolvedVersion, LibraryType.Package))
+                                    .ToList();
+
+                                // add lock file libraries into RemoteWalkContext so that it can be used during restore graph generation
+                                contextForProject.LockFileLibraries.Add(new LockFileCacheKey(target.TargetFramework, target.RuntimeIdentifier), libraries);
```

Comment:
I don't see a reason why you need a mapping for the libraries.
You only need a set of all the libraries. You never make use of the mapping.

Reply:
We need this mapping to resolve dependencies which happens in `ResolverUtility`